OrbitVulkanLayer/VulkanLayerControllerTest.cpp

```diff
@@ -0,0 +1,775 @@
+  MOCK_METHOD(PFN_vkResetCommandPool, ResetCommandPool, (VkDevice));
+  MOCK_METHOD(PFN_vkAllocateCommandBuffers, AllocateCommandBuffers, (VkDevice));
+  MOCK_METHOD(PFN_vkFreeCommandBuffers, FreeCommandBuffers, (VkDevice));
+  MOCK_METHOD(PFN_vkBeginCommandBuffer, BeginCommandBuffer, (VkCommandBuffer));
+  MOCK_METHOD(PFN_vkEndCommandBuffer, EndCommandBuffer, (VkCommandBuffer));
+  MOCK_METHOD(PFN_vkResetCommandBuffer, ResetCommandBuffer, (VkCommandBuffer));
+  MOCK_METHOD(PFN_vkGetDeviceQueue, GetDeviceQueue, (VkDevice));
+  MOCK_METHOD(PFN_vkGetDeviceQueue2, GetDeviceQueue2, (VkDevice));
+  MOCK_METHOD(PFN_vkQueueSubmit, QueueSubmit, (VkQueue));
+  MOCK_METHOD(PFN_vkQueuePresentKHR, QueuePresentKHR, (VkQueue));
+  MOCK_METHOD(PFN_vkCmdBeginDebugUtilsLabelEXT, CmdBeginDebugUtilsLabelEXT, (VkCommandBuffer));
+  MOCK_METHOD(PFN_vkCmdEndDebugUtilsLabelEXT, CmdEndDebugUtilsLabelEXT, (VkCommandBuffer));
+  MOCK_METHOD(PFN_vkCmdDebugMarkerBeginEXT, CmdDebugMarkerBeginEXT, (VkCommandBuffer));
+  MOCK_METHOD(PFN_vkCmdDebugMarkerEndEXT, CmdDebugMarkerEndEXT, (VkCommandBuffer));
+  MOCK_METHOD(bool, IsDebugUtilsExtensionSupported, (VkCommandBuffer));
+  MOCK_METHOD(bool, IsDebugMarkerExtensionSupported, (VkCommandBuffer));
+};
+
+class MockDeviceManager {
+ public:
+  explicit MockDeviceManager(MockDispatchTable* /*dispatch_table*/) {}
+  MOCK_METHOD(void, TrackLogicalDevice, (VkPhysicalDevice, VkDevice));
+  MOCK_METHOD(void, UntrackLogicalDevice, (VkDevice));
+};
+
+class MockQueueManager {
+ public:
+  MOCK_METHOD(void, TrackQueue, (VkQueue, VkDevice));
+  MOCK_METHOD((VkDevice), GetDeviceOfQueue, (VkQueue));
+};
+
+class MockTimerQueryPool {
+ public:
+  explicit MockTimerQueryPool(MockDispatchTable* /*dispatch_table*/, uint32_t /*num_slots*/) {}
+  MOCK_METHOD(void, InitializeTimerQueryPool, (VkDevice));
+};
+
+struct Color {
+  float red;
+  float green;
+  float blue;
+  float alpha;
+};
+
+class MockSubmissionTracker {
+ public:
+  struct QueueSubmission {};
+
+  explicit MockSubmissionTracker(MockDispatchTable* /*dispatch_table*/,
+                                 MockTimerQueryPool* /*timer_query_pool*/,
+                                 MockDeviceManager* /*device_manager*/, uint32_t /*max_depth*/) {}
+  MOCK_METHOD(void, SetVulkanLayerProducer, (VulkanLayerProducer*));
+  MOCK_METHOD(void, ResetCommandPool, (VkCommandPool));
+  MOCK_METHOD(void, TrackCommandBuffers,
+              (VkDevice, VkCommandPool, const VkCommandBuffer*, uint32_t));
+  MOCK_METHOD(void, UntrackCommandBuffers,
+              (VkDevice, VkCommandPool, const VkCommandBuffer*, uint32_t));
+  MOCK_METHOD(void, MarkCommandBufferBegin, (VkCommandBuffer));
+  MOCK_METHOD(void, MarkCommandBufferEnd, (VkCommandBuffer));
+  MOCK_METHOD(void, ResetCommandBuffer, (VkCommandBuffer));
+  MOCK_METHOD(std::optional<QueueSubmission>, PersistCommandBuffersOnSubmit,
+              (uint32_t, const VkSubmitInfo* submits));
+  MOCK_METHOD(bool, PersistDebugMarkersOnSubmit,
```

Comment:
Super minor thing I noticed: this is originally returning a `void`. You don't need the mock to return a `bool`, right?

Reply:
Very good catch! Thanks!